### PR TITLE
[BugFix] Fix condition will always be true error when check functionset name in mv

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by caneGuy** [comment]
@Astralidea @ABingHuang PTAL thanks

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java|0|1|00.00%|[680]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java [hunk 1]
```diff
@@ -677,7 +677,7 @@ private boolean aggFunctionsMatchAggColumns(Map<String, Integer> columnToIds,
         if (candidateIndexMeta.getKeysType() == KeysType.AGG_KEYS && !queryExprList.stream()
                 .filter(queryExpr -> {
                     String fnName = queryExpr.getFnName();
-                    return !(fnName.equals(FunctionSet.COUNT) && fnName.equals(FunctionSet.SUM))
+                    return !(fnName.equals(FunctionSet.COUNT) || fnName.equals(FunctionSet.SUM))
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java:680]: better to add UT cover this case

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java|7|7|100.00%|[]|

**Comment by caneGuy** [comment]
run starrocks_admit_test

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java [hunk 2]
```diff
@@ -669,25 +669,24 @@ private boolean aggFunctionsMatchAggColumns(Map<String, Integer> columnToIds,
...
             boolean match = false;
             for (CallOperator mvExpr : mvColumnExprList) {
                 if (isMVMatchAggFunctions(indexId, queryExpr, mvExpr, mvColumnExprList, keyColumns, aggregateColumns)) {
                     match = true;
+                    /*
+                    If all the columns in the query are key column and materialized view is AGG_KEYS.
+                    We should skip this query for given materialized view.
+                    ISSUE-6984: https://github.com/StarRocks/starrocks/issues/6984
+                    */
+                    if (candidateIndexMeta.getKeysType() == KeysType.AGG_KEYS && !queryExprList.stream()
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java:682]: why move it here?
what is queryExpr0 for? seems useless?

**Review by ABingHuang — CHANGES_REQUESTED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRule.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**